corehq/sql_db/connections.py

```diff
@@ -104,26 +116,74 @@ def dispose_all(self):
-            UCR_ENGINE_ID: settings.UCR_DATABASE_URL,
-        }
-        if hasattr(settings, 'ICDS_UCR_DATABASE_ALIAS') and settings.ICDS_UCR_DATABASE_ALIAS in settings.DATABASES:
-            db_connection_map[ICDS_UCR_ENGINE_ID] = \
-                "postgresql+psycopg2://{USER}:{PASSWORD}@{HOST}:{PORT}/{NAME}".format(
-                    **settings.DATABASES[settings.ICDS_UCR_DATABASE_ALIAS]
-                )
-        if hasattr(
-            settings, 'ICDS_UCR_TEST_DATABASE_ALIAS'
-        ) and settings.ICDS_UCR_TEST_DATABASE_ALIAS in settings.DATABASES:
-            db_connection_map[
-                ICDS_TEST_UCR_ENGINE_ID
-            ] = "postgresql+psycopg2://{USER}:{PASSWORD}@{HOST}:{PORT}/{NAME}".format(
-                **settings.DATABASES[settings.ICDS_UCR_TEST_DATABASE_ALIAS]
-            )
-        for custom_engine_id, custom_db_url in settings.CUSTOM_DATABASES:
-            db_connection_map[custom_engine_id] = custom_db_url
-        return db_connection_map.get(engine_id, settings.SQL_REPORTING_DATABASE_URL)
+        connection_string = self.db_connection_map.get(engine_id)
+        return connection_string or self.db_connection_map['default']
+
+    def _populate_connection_map(self):
+        reporting_db_config = self._get_reporting_db_config()
+        if not reporting_db_config:
+            self._populate_from_legacy_settings()
+        else:
+            for engine_id, db_config in reporting_db_config.items():
+                write_db = db_config
+                read = None
+                if isinstance(db_config, dict):
+                    write_db = db_config['WRITE']
+                    read = db_config['READ']
+                    for db_alias, weighting in read:
+                        assert isinstance(weighting, int), 'weighting must be int'
+                        assert db_alias in settings.DATABASES, db_alias
+
+                self._add_django_db(engine_id, write_db)
+                if read:
+                    self.read_database_mapping[engine_id] = []
+                    for read_db, weighting in read:
+                        assert read_db == write_db or read_db not in self.db_connection_map, read_db
+                        self.read_database_mapping[engine_id].extend([read_db] * weighting)
+                        if read_db != write_db:
+                            self._add_django_db(read_db, read_db)
+
+        if DEFAULT_ENGINE_ID not in self.db_connection_map:
+            self._add_django_db(DEFAULT_ENGINE_ID, 'default')
+        if UCR_ENGINE_ID not in self.db_connection_map:
+            self._add_django_db(UCR_ENGINE_ID, 'default')
+
+        self._add_django_db_from_settings_key(ICDS_UCR_ENGINE_ID, 'ICDS_UCR_DATABASE_ALIAS')
+        self._add_django_db_from_settings_key(ICDS_TEST_UCR_ENGINE_ID, 'ICDS_UCR_TEST_DATABASE_ALIAS')
+
+    def _populate_from_legacy_settings(self):
+        sql_reporting_db_url = getattr(settings, 'SQL_REPORTING_DATABASE_URL', None)
```

Comment:
It looks like you removed this setting, and it isn't defined anywhere in ansible, so I think it can be removed.

Reply:
ha, nice: 2da4e1d